Release Scene Builder kit to maven central

JReleaser needs to be updated to also release kit to Maven Central. Currently the kit jar file is only pushed to Github Releases and AWS S3.

In addition to this, we also need to push snapshots to maven central as well.

Thanks, @aalmiray, I've followed your instructions, and created the PR #759.
 
In order to test it, which would require merging into main, I'm running a test branch here: https://github.com/jperedadnr/scenebuilder/tree/test-publish (see https://github.com/jperedadnr/scenebuilder/actions/runs/11392044901), and based on it I've done some more changes.

However, I'm not sure if there is an issue with the password/token I'm using (via secret), as I keep getting:
```
[INFO]      [nexus2]  - com/gluonhq/scenebuilder/kit/24.0.0-SNAPSHOT/kit-24.0.0-20241017.195447-1-javadoc.jar
[DEBUG]     [nexus2]  - deploying kit-24.0.0-20241017.195447-1-javadoc.jar as /com/gluonhq/scenebuilder/kit/24.0.0-SNAPSHOT/kit-24.0.0-20241017.195447-1-javadoc.jar
[DEBUG]     [nexus2] url: https://s01.oss.sonatype.org/content/repositories/snapshots/com/gluonhq/scenebuilder/kit/24.0.0-SNAPSHOT/kit-24.0.0-20241017.195447-1-javadoc.jar
[DEBUG]     [nexus2] opening connection
[DEBUG]     [nexus2] configuring connection
[DEBUG]     [nexus2] Accept: [*/*]
[DEBUG]     [nexus2] Expect: [100-continue]
[DEBUG]     [nexus2] User-Agent: [JReleaser/1.14.0]
[DEBUG]     [nexus2] Content-Type: [application/java-archive]
[DEBUG]     [nexus2] sending data
[DEBUG]     [nexus2] handling response
server replied with 401: Content access is protected by token
[ERROR]     [nexus2]  x kit-24.0.0-20241017.195447-1-javadoc.jar
org.jreleaser.sdk.nexus2.Nexus2Exception: Error when deploying artifact kit-24.0.0-20241017.195447-1-javadoc.jar
	at org.jreleaser.sdk.nexus2.Nexus2.fail(Nexus2.java:305)
	at org.jreleaser.sdk.nexus2.Nexus2.deploy(Nexus2.java:296)
	at org.jreleaser.sdk.nexus2.Nexus2MavenDeployer.uploadArtifacts(Nexus2MavenDeployer.java:250)
	at org.jreleaser.sdk.nexus2.Nexus2MavenDeployer.deploy(Nexus2MavenDeployer.java:160)
...
```

Oh, you are right about using the legacy server https://oss.sonatype.org instead of https://s01.oss.sonatype.org

It is confusing, since we can sign in in both with the same old Nexus user/password, and therefore I thought the new user/token secrets were the same for both. But now I see there are different, and I have now a new pair of secrets for s01.

Running again with these and s01, I get: 

```
Caused by: org.jreleaser.model.spi.deploy.DeployException: Unexpected error when deploying target/staging-deploy/com/gluonhq/scenebuilder/kit/24.0.0-SNAPSHOT/kit-24.0.0-20241018.163838-1-javadoc.jar: Error when deploying artifact kit-24.0.0-20241018.163838-1-javadoc.jar
	at org.jreleaser.sdk.nexus2.Nexus2MavenDeployer.uploadArtifacts(Nexus2MavenDeployer.java:253)
	at org.jreleaser.sdk.nexus2.Nexus2MavenDeployer.deploy(Nexus2MavenDeployer.java:160)
	at org.jreleaser.engine.deploy.maven.ProjectMavenDeployer.deploy(ProjectMavenDeployer.java:55)
	at org.jreleaser.engine.deploy.maven.MavenDeployers.deploy(MavenDeployers.java:167)
	... 33 more
Caused by: org.jreleaser.sdk.nexus2.Nexus2Exception: Error when deploying artifact kit-24.0.0-20241018.163838-1-javadoc.jar
	at org.jreleaser.sdk.nexus2.Nexus2.fail(Nexus2.java:305)
	at org.jreleaser.sdk.nexus2.Nexus2.deploy(Nexus2.java:296)
	at org.jreleaser.sdk.nexus2.Nexus2MavenDeployer.uploadArtifacts(Nexus2MavenDeployer.java:250)
	... 36 more
Caused by: org.jreleaser.model.spi.upload.UploadException: Got 403 reason: Forbidden
	at org.jreleaser.sdk.commons.ClientUtils.uploadFile(ClientUtils.java:327)
	at org.jreleaser.sdk.commons.ClientUtils.putFile(ClientUtils.java:253)
	at org.jreleaser.sdk.nexus2.Nexus2.deploy(Nexus2.java:288)
	... 37 more
```

Why can we publish to old Sonatype with maven-deploy-plugin, but not to s01 with JReleaser?

--- a/.github/workflows/early-access.yml
+++ b/.github/workflows/early-access.yml
@@ -149,14 +149,10 @@ jobs:
         run: |
           mvn -B -ntp -pl :parent -DartifactsDir=artifacts jreleaser:release
 
-      - name: JReleaser output
-        if: always()
-        uses: actions/upload-artifact@v4
-        with:
-          name: jreleaser-logs
-          path: |
-            target/jreleaser/trace.log
-            target/jreleaser/output.properties
+      - name: Deploy parent pom and Kit to local staging
+        run: |
+          mvn -B -ntp deploy -pl :parent -Dmaven.test.skip=true
+          mvn -B -ntp deploy -fkit -Dmaven.test.skip=true
 
       - name: Publish Kit with JReleaser
         env:
@@ -167,5 +163,13 @@ jobs:
           JRELEASER_GPG_PUBLIC_KEY: ${{ secrets.JRELEASER_GPG_PUBLIC_KEY }}
           JRELEASER_GPG_SECRET_KEY: ${{ secrets.JRELEASER_GPG_SECRET_KEY }}
         run: |
-          mvn deploy -fkit
           mvn -B -ntp -pl :parent jreleaser:deploy
+
+      - name: JReleaser output
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: jreleaser-logs
+          path: |
+            target/jreleaser/trace.log
+            target/jreleaser/output.properties

--- a/.github/workflows/early-access.yml
+++ b/.github/workflows/early-access.yml
@@ -157,3 +157,15 @@ jobs:
           path: |
             target/jreleaser/trace.log
             target/jreleaser/output.properties
+
+      - name: Publish Kit with JReleaser
+        env:
+          JRELEASER_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          JRELEASER_NEXUS2_USERNAME: ${{ secrets.JRELEASER_NEXUS2_USERNAME }}
+          JRELEASER_NEXUS2_PASSWORD: ${{ secrets.JRELEASER_NEXUS2_TOKEN }}
+          JRELEASER_GPG_PASSPHRASE: ${{ secrets.JRELEASER_GPG_PASSPHRASE }}
+          JRELEASER_GPG_PUBLIC_KEY: ${{ secrets.JRELEASER_GPG_PUBLIC_KEY }}
+          JRELEASER_GPG_SECRET_KEY: ${{ secrets.JRELEASER_GPG_SECRET_KEY }}
+        run: |
+          mvn deploy -fkit
+          mvn -B -ntp -pl :parent jreleaser:deploy

--- a/gluon-plugin/pom.xml
+++ b/gluon-plugin/pom.xml
@@ -15,6 +15,13 @@
         <gluon.attach.version>4.0.21</gluon.attach.version>
     </properties>
 
+    <repositories>
+        <repository>
+            <id>gluon-nexus</id>
+            <url>https://nexus.gluonhq.com/nexus/content/repositories/releases/</url>
+        </repository>
+    </repositories>
+
     <dependencies>
         <dependency>
             <groupId>com.gluonhq.scenebuilder</groupId>

--- a/kit/pom.xml
+++ b/kit/pom.xml
@@ -3,12 +3,19 @@
     <modelVersion>4.0.0</modelVersion>
     <artifactId>kit</artifactId>
     <name>Scene Builder Kit</name>
-    
+    <description>Scene Builder is a visual, drag n drop, layout tool for designing JavaFX application user interfaces</description>
+    <inceptionYear>2012</inceptionYear>
+    <url>https://github.com/gluonhq/scenebuilder</url>
+
     <parent>
         <groupId>com.gluonhq.scenebuilder</groupId>
         <artifactId>parent</artifactId>
         <version>24.0.0-SNAPSHOT</version>
     </parent>
+
+    <properties>
+        <altDeploymentRepository>local::file:./target/staging-deploy</altDeploymentRepository>
+    </properties>
 
     <dependencies>
         <!-- JavaFX -->
@@ -84,16 +91,52 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-jar-plugin</artifactId>
-                <version>3.2.0</version>
+                <version>3.4.2</version>
                 <configuration>
                     <finalName>scenebuilder-kit-${project.version}</finalName>
                     <outputDirectory>target/lib</outputDirectory>
-                    <archive>
-                        <manifestEntries>
-                            <Automatic-Module-Name>com.gluonhq.scenebuilder.kit</Automatic-Module-Name>
-                        </manifestEntries>
-                    </archive>
                 </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-deploy-plugin</artifactId>
+                <version>3.1.3</version>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-javadoc-plugin</artifactId>
+                <version>3.10.1</version>
+                <executions>
+                    <execution>
+                        <id>attach-javadocs</id>
+                        <goals>
+                            <goal>jar</goal>
+                        </goals>
+                        <configuration>
+                            <attach>true</attach>
+                            <doclint>none</doclint>
+                            <jarOutputDirectory>${project.build.directory}/lib</jarOutputDirectory>
+                            <outputDirectory>${project.build.directory}/lib</outputDirectory>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-source-plugin</artifactId>
+                <version>3.3.1</version>
+                <executions>
+                    <execution>
+                        <id>attach-sources</id>
+                        <goals>
+                            <goal>jar</goal>
+                        </goals>
+                        <configuration>
+                            <attach>true</attach>
+                            <outputDirectory>${project.build.directory}/lib</outputDirectory>
+                        </configuration>
+                    </execution>
+                </executions>
             </plugin>
         </plugins>
     </build>

--- a/kit/pom.xml
+++ b/kit/pom.xml
@@ -13,10 +13,6 @@
         <version>24.0.0-SNAPSHOT</version>
     </parent>
 
-    <properties>
-        <altDeploymentRepository>local::file:./target/staging-deploy</altDeploymentRepository>
-    </properties>
-
     <dependencies>
         <!-- JavaFX -->
         <dependency>

--- a/kit/pom.xml
+++ b/kit/pom.xml
@@ -99,11 +99,6 @@
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-deploy-plugin</artifactId>
-                <version>3.1.3</version>
-            </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>3.10.1</version>
                 <executions>

--- a/pom.xml
+++ b/pom.xml
@@ -349,8 +349,9 @@
                                 <nexus2>
                                     <maven-central>
                                         <active>ALWAYS</active>
-                                        <url>https://s01.oss.sonatype.org/service/local</url>
-                                        <snapshotUrl>https://s01.oss.sonatype.org/content/repositories/snapshots</snapshotUrl>
+                                        <url>https://oss.sonatype.org/service/local</url>
+                                        <snapshotSupported>true</snapshotSupported>
+                                        <snapshotUrl>https://oss.sonatype.org/content/repositories/snapshots</snapshotUrl>
                                         <closeRepository>true</closeRepository>
                                         <releaseRepository>false</releaseRepository>
                                         <stagingRepositories>target/staging-deploy</stagingRepositories>

--- a/pom.xml
+++ b/pom.xml
@@ -101,23 +101,12 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-compiler-plugin</artifactId>
-                    <version>3.8.1</version>
+                    <version>3.13.0</version>
                     <configuration>
                         <useIncrementalCompilation>false</useIncrementalCompilation>
                     </configuration>
                 </plugin>
 
-                <!-- Deploy to Maven Central -->
-                <plugin>
-                    <groupId>org.apache.maven.plugins</groupId>
-                    <artifactId>maven-deploy-plugin</artifactId>
-                    <version>3.0.0-M1</version>
-                    <configuration>
-                        <skip>true</skip>
-                    </configuration>
-                    <inherited>false</inherited>
-                </plugin>
-                
                 <!-- Enforce code formatting and style -->
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
@@ -202,7 +191,7 @@
             <plugin>
                 <groupId>org.jreleaser</groupId>
                 <artifactId>jreleaser-maven-plugin</artifactId>
-                <version>1.5.1</version>
+                <version>1.14.0</version>
                 <inherited>false</inherited>
                 <configuration>
                     <jreleaser>
@@ -355,6 +344,24 @@
                                 </artifact>
                             </artifacts>
                         </files>
+                        <signing>
+                            <active>ALWAYS</active>
+                            <armored>true</armored>
+                        </signing>
+                        <deploy>
+                            <maven>
+                                <nexus2>
+                                    <maven-central>
+                                        <active>ALWAYS</active>
+                                        <url>https://s01.oss.sonatype.org/service/local</url>
+                                        <snapshotUrl>https://s01.oss.sonatype.org/content/repositories/snapshots/</snapshotUrl>
+                                        <closeRepository>true</closeRepository>
+                                        <releaseRepository>false</releaseRepository>
+                                        <stagingRepositories>target/staging-deploy</stagingRepositories>
+                                    </maven-central>
+                                </nexus2>
+                            </maven>
+                        </deploy>
                     </jreleaser>
                 </configuration>
             </plugin>

--- a/pom.xml
+++ b/pom.xml
@@ -30,18 +30,8 @@
         <project.github.repository>gluonhq/scenebuilder</project.github.repository>
         <repository.url>github.com:${project.github.repository}</repository.url>
         <jreleaser.files.active>RELEASE</jreleaser.files.active>
+        <altDeploymentRepository>local::file:./target/staging-deploy</altDeploymentRepository>
     </properties>
-
-    <repositories>
-        <repository>
-            <id>sonatype-snapshots</id>
-            <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
-        </repository>
-        <repository>
-            <id>gluon-nexus</id>
-            <url>https://nexus.gluonhq.com/nexus/content/repositories/releases/</url>
-        </repository>
-    </repositories>
 
     <dependencyManagement>
         <dependencies>
@@ -132,6 +122,12 @@
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-resources-plugin</artifactId>
                     <version>3.2.0</version>
+                </plugin>
+
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-deploy-plugin</artifactId>
+                    <version>3.1.3</version>
                 </plugin>
 
                 <!-- Ensure that available tests are executed -->
@@ -354,7 +350,7 @@
                                     <maven-central>
                                         <active>ALWAYS</active>
                                         <url>https://s01.oss.sonatype.org/service/local</url>
-                                        <snapshotUrl>https://s01.oss.sonatype.org/content/repositories/snapshots/</snapshotUrl>
+                                        <snapshotUrl>https://s01.oss.sonatype.org/content/repositories/snapshots</snapshotUrl>
                                         <closeRepository>true</closeRepository>
                                         <releaseRepository>false</releaseRepository>
                                         <stagingRepositories>target/staging-deploy</stagingRepositories>

